tests/test_parser.py

```diff
@@ -28,6 +28,116 @@
 instruction = instruction  # pylint: disable=self-assigning-variable
```

Comment:
codeQL trips on this line. apparently you can ignore codeQL errors using an UI? (but i'm not sure where I would find this...).